Automattic/sensei
# Learning Mode - Support multiple custom block templates for a single post type.

Fixes #5538 

### Changes proposed in this Pull Request

* Adds the ability to store multiple custom block template for a single  post type.

Currently when the customized version of the lesson/quiz templates are stored in the database it is simply stored as a lesson or quiz template. There is nothing that tells us which Learning Mode template is used to create that customization. Whether it is Default, Modern, Video or Video Full template, we don't know.

The only thing we do know is that it's either `lesson` or `quiz` that is used as the `post_name` of the custom block template.

So, in order to distinguish between a Default lesson template and Modern lesson template, in this PR we are appending the name of the LM block template into the `post_name` of the block template in the db. This allows us to detect which LM template was used to create those custom block templates that are stored in the database.

### Testing instructions

* Make sure to git checkout `feature/learning-mode` in `sensei-pro` repo. And git checkout this branch for `sensei` repo.
1. Choose any template in Senei -> Settings -> Appearance.
2. Edit that template in Side Editor and save.
3. Confirm that changes are reflected on the frontend.
4. Go back to Sensei -> Settings -> Appearance and choose another LM template.
5. Confirm that frontend now uses the new template that you chose in the Sensei Settings.
6. Edit that the current active template in site editor and confirm the changes are reflected on the frontend.
7. Go to Sensei Settings again and change back to the first template you choose in step 1.
8. Confirm that the frontend now renders your customized template that you saved in step 2.

Bonus points: Try out playing with different block based and non block based themes.


### Screenshot / Video

https://user-images.githubusercontent.com/2578542/189830620-6e020d1f-4174-420c-857e-b6b09bb4b48e.mp4
